src/ApiHandler.ts

```diff
@@ -139,6 +125,72 @@ export default class ApiHandler {
 			}
 		}
 
+		const perByteFee = api.consts.transactionPayment.transactionByteFee;
+		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
+		const coefficients = api.consts.transactionPayment.weightToFee[0];
+		const coeffFrac = coefficients.coeff_frac;
+		const coeffInt = coefficients.coeff_integer;
+		
+		// TODO: We assume that the polynomial has one positive term of degree one.
+		// const degree = coefficients.degree;
+		// const negative = coefficients.negative;
+
+		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			if (!extrinsics[idx].paysFee) {
+				continue;
+			}
+
+			try {
+				if (api.runtimeVersion.specName.toString() === 'kusama') {
+					extrinsics[idx].info = api.createType('RuntimeDispatchInfo',await api.rpc.payment.queryInfo(block.extrinsics[idx].toHex(), parentHash));
+					continue;
+				}
+
+				const xtEvents = extrinsics[idx].events;
+				const success = 'system.ExtrinsicSuccess';
+				const failure = 'system.ExtrinsicFailed';
+				const completedEvent = xtEvents.find((event) => event.method === success || event.method === failure);
+				if (!completedEvent) {
+					extrinsics[idx].info = { error: 'Unable to find success or failure event for extrinsic' };
+					continue;
+				}
+
+				const completedData = completedEvent.data;
+				if (!completedData) {
+					extrinsics[idx].info = { error: 'Success or failure event for extrinsic does not contain expected data' };
+					continue;
+				}
+
+				// both ExtrinsicSuccess and ExtrinsicFailed events have DispatchInfo types as their final arg
+				const weightInfo = completedData[completedData.length - 1] as DispatchInfo;
+				if (!weightInfo.weight) {
+					extrinsics[idx].info = { error: 'Success or failure event for extrinsic does not specify weight' };
+					continue;
+				}
+
+				const weight = weightInfo.weight;
+				const encodedLength = block.extrinsics[idx].encodedLength;
+				const multiplier = await api.query.transactionPayment.nextFeeMultiplier.at(parentHash);
```

Comment:
This can (and should, since it's an RPC call) also come out of the for loop as it's constant for all extrinsics in the block.